pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -500,7 +500,28 @@ def set_distributed_mode(self, distributed_backend: Optional[str] = None):
             else:
                 rank_zero_warn('You are running on single node with no parallelization, so distributed has no effect.')
                 self._distrib_type = None
-
+        
+        # finished configuring self._distrib_type, check ipython environment
+        ipython_incompatible_distrib_types = [DistributedType.DDP, DistributedType.DDP2, DistributedType.HOROVOD]
```

Comment:
It should be simpler to list the compatible ones as only `DP`, `DDP_SPAWN ` and `DDP_SHARDED_SPAWN ` should be.

```
class DistributedType(LightningEnum):
    """ Define type of ditributed computing.

    >>> # you can math the type with string
    >>> DistributedType.DDP == 'ddp'
    True
    >>> # which is case invariant
    >>> DistributedType.DDP2 in ('ddp2', )
    True
    """
    DP = 'dp'
    DDP = 'ddp'
    DDP2 = 'ddp2'
    DDP_SPAWN = 'ddp_spawn'
    HOROVOD = 'horovod'
    DDP_SHARDED = 'ddp_sharded'
    DDP_SHARDED_SPAWN = 'ddp_sharded_spawn'
    RPC_SEQUENTIAL_PLUGIN = 'rpc_sequential'
```

Reply:
Make sense, so I added an attribute `self.ipython_compatible_distrib_types` to record compatibility in case of future compatibility improvements